skimage/restoration/tests/test_denoise.py

```diff
@@ -182,6 +183,18 @@ def test_denoise_bilateral_2d():
     assert_(out1[30:45, 5:15].std() > out2[30:45, 5:15].std())
 
 
+@pytest.mark.parametrize('dtype', [np.float16, np.float32, np.double])
+def test_denoise_bilateral_types(dtype):
+    img = checkerboard_gray.copy()[:50, :50]
+    # add some random noise
+    img += 0.5 * img.std() * np.random.rand(*img.shape)
+    img = np.clip(img, 0, 1)
+
+    # make sure that it can actually process the float32 types
```

Comment:
This should probably be changed to say `float32` to `all valid float` - I can take care of that if that's fine with you